packages/core/src/util/spanFilter.js

```diff
@@ -5,36 +5,100 @@
 'use strict';
 
 // List of span types to allowed to ignore
-const IGNORABLE_SPAN_TYPES = ['redis', 'dynamodb'];
+// 'kafka_placeholder' used as a temporary span type for testing, until the Kafka support is implemented.
+const IGNORABLE_SPAN_TYPES = ['redis', 'dynamodb', 'kafka_placeholder'];
 
 /**
  * @param {import('../core').InstanaBaseSpan} span
- * @param {import('../tracing').IgnoreEndpoints} endpoints
+ * @param {import('../tracing').IgnoreEndpointConfig} ignoreconfig
  * @returns {boolean}
  */
-function shouldIgnore(span, endpoints) {
-  // Skip if the span type is not in the ignored list
-  if (!IGNORABLE_SPAN_TYPES.includes(span.n)) {
+function matchEndpoints(span, ignoreconfig) {
+  const spanEndpoints = Array.isArray(span.data[span.n]?.endpoints)
+    ? span.data[span.n].endpoints
+    : [span.data[span.n]?.endpoints].filter(Boolean);
+
+  if (!spanEndpoints.length) {
     return false;
   }
-  const endpoint = endpoints[span.n];
-  if (!endpoint) return false;
 
-  // The endpoint already been converted to lowercase during parsing, so here we are normalizing
-  // the operation for case-insensitive comparison
-  const operation = span.data?.[span.n]?.operation?.toLowerCase();
-  if (!operation) return false;
+  // If * is present, any endpoint will match.
+  if (ignoreconfig.endpoints.includes('*')) {
+    return true;
+  }
 
-  // We support both array and string formats for endpoints, but the string format is not shared publicly.
-  if (Array.isArray(endpoint)) {
-    return endpoint.some(op => op.toLowerCase() === operation);
+  return spanEndpoints.some((/** @type {string} */ endpoint) =>
```

Comment:
Please remove these types here and use:

  /**
   * @type {string[]}
   */
     const spanEndpoints = Array.isArray(span.data[span.n]?.endpoints)
    ? span.data[span.n].endpoints
    : [span.data[span.n]?.endpoints].filter(Boolean);